homeassistant/components/light/yeelightsunflower.py

```diff
@@ -0,0 +1,124 @@
+"""
+Support for Yeelight Sunflower colour bulbs (not Yeelight Blue or WiFi).
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.yeelight-sunflower
+Uses the yeelightsunflower library:
+https://github.com/lindsaymarkward/python-yeelight-sunflower
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.components.light import (Light,
+                                            ATTR_RGB_COLOR, SUPPORT_RGB_COLOR,
+                                            ATTR_BRIGHTNESS,
+                                            SUPPORT_BRIGHTNESS,
+                                            PLATFORM_SCHEMA)
+from homeassistant.const import CONF_HOST
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['yeelightsunflower==0.0.2']
+SUPPORT_YEELIGHT_SUNFLOWER = (SUPPORT_BRIGHTNESS | SUPPORT_RGB_COLOR)
+
+_LOGGER = logging.getLogger(__name__)
+
+# Validate the user's configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Yeelight Sunflower Light platform."""
+    import yeelightsunflower
+
+    # Assign configuration variables.
+    # The configuration check takes care they are present.
+    host = config.get(CONF_HOST)
+
+    # Setup connection with Yeelight Sunflower hub
+    hub = yeelightsunflower.Hub(host)
+
+    # Verify that hub is responsive
+    if not hub.available:
+        _LOGGER.error('Could not connect to Yeelight Sunflower hub')
+        return False
+
+    # Add devices
+    add_devices(SunflowerBulb(light) for light in hub.get_lights())
+
+
+class SunflowerBulb(Light):
+    """Representation of a Yeelight Sunflower Light."""
+
+    def __init__(self, light):
+        """Initialize a Yeelight Sunflower bulb."""
+        self._light = light
+        self._available = light.available
+        self._brightness = light.brightness
+        self._is_on = light.is_on
+        self._rgb_color = light.rgb_color
+
+    @property
+    def unique_id(self):
+        """Return the ID of this light."""
+        return "{}.{}".format(self.__class__, self._light.zid)
```

Comment:
OK. They will be different, but I was just filling in all of the methods I saw in the parent that made sense. If name covers it, then I'll take out unique_id.